### ISSUE TITLE
Out of memory error when downloading large dataset

### ISSUE THREAD

**Issue by ethanwhite** [weecology/retriever#290]
When trying to install the new eBird Observational Dataset (#286) I run into an out of memory error on my laptop.

```
ethan@oryx:~/$ retriever install postgres eBirdOD
=> Installing eBird Observation Dataset
Creating database eBirdOD...
Couldn't create database (schema "ebirdod" already exists
). Trying to continue anyway.
Downloading CLOEODDATA.05192014.1...
out of memory
```

This is presumably because `download_file()` in `engine.py` is reading in the entire file and then writing it. This means that we need to read and write more manageable chunks.

**Comment by bendmorris** [comment]
Definitely. Beware of the edge case - download_file is also doing conversion of line endings to unix-style. If you read chunks, you might split a "\r\n" in half which would result in a double break.

**Comment by ethanwhite** [comment]
Yeah, I was worried about that. Thoughts on #291?

**Comment by bendmorris** [comment]
Nevermind, just noticed this: https://github.com/weecology/retriever/issues/291

I think 'U' mode will work correctly if reading line by line, not sure what would happen if reading fixed-size blocks. Reading fairly large blocks is going to be faster than reading lines. Should probably test whether this affects performance downloading different datasets before going with one approach.

**Comment by ethanwhite** [comment]
If 'U' works then I was thinking we'd just skip cleaning the line endings in `download_file()`. Download the whole thing as it is, in chunks, and just use 'U' whenever we're opening things after the download. I those post-download opens are already happening either line by line or with the entire file, never in blocks at the moment. Does this make sense?

**Comment by bendmorris** [comment]
I think so, sounds good to me.

**Comment by ethanwhite** [comment]
Fixed by @Deborah-Digges in f4d34a0f3ab399775d6baaa160284a8e974d293e.
